dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java

```diff
@@ -142,41 +102,55 @@ public Category find(final String id, final User user,
 		return permissionAPI.filterCollection(categories, PermissionAPI.PERMISSION_READ, respectFrontendRoles, user);
 	}
 
+	/**
+	 * Saves a category
+	 *
+	 * When saving a new category the parent should be passed to the API
+	 * to check if the user has permissions to add children to the parent
+	 * and the parent will be associated to the passed category
+	 *
+	 * @param parent Parent can be null if saving an top level category
+	 * @param category Category to be saved
+	 * @param user user that is performing the save
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
 	@WrapInTransaction
 	public void save(final Category parent,
-					 Category object,
+					 Category category,
 					 final User user,
 					 final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
 
         // Checking that we have a unique key.
-	    object = checkUniqueKey(object, user);
-
-	    boolean isANewCategory = false;
-
-		//Checking permissions
-		if(InodeUtils.isSet(object.getInode()) || parent == null) {
-			//Object is not new or is a top level category
-			//if it is a new top level category the user should be a cms administrator
-			// and that's checked in the permissions api
-			 if(!com.dotmarketing.business.APILocator.getRoleAPI().doesUserHaveRole(user, com.dotmarketing.business.APILocator.getRoleAPI().loadCMSAdminRole().getId())){
-              if(!permissionAPI.doesUserHavePermission(object, PermissionAPI.PERMISSION_EDIT, user, respectFrontendRoles))
-				throw new DotSecurityException("User doesn't have permission to edit the category = " + object.getInode());
-			 }
-		} else {
-			//Object is new and a parent was provided so we check in the parent permissions
-			if(!permissionAPI.doesUserHavePermission(parent, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user, respectFrontendRoles))
-				throw new DotSecurityException("User doesn't have permission to save this category = " +
-						object.getInode() + " having as parent the category = " + parent.getInode());
+	    category = checkUniqueKey(category, user);
 
-			isANewCategory = true;
+	    boolean isANewCategory = UtilMethods.isNotSet(category.getInode());

```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6178451)